architecture/design/smart-driver.md

```diff
@@ -0,0 +1,233 @@
+<!-- Output copied to clipboard! -->
+
+
+<h1>Smart JDBC driver for YSQL</h1>
+
+
+<h2>Motivation</h2>
+
+
+The current Yugabyte JDBC driver is built on the Postgres JDBC driver. It works fine with YugabyteDB as YugabyteDB is wire compatible with PostgreSQL. However, YugabyteDB being a distributed database can use some smarts to not only increase performance but also obviate the need of external load balancers like HAProxy, pgpool etc. Elaborating on the four key motivations.
+
+<h4>Load Balancing</h4>
```

Comment:
As discussed, lets make this `Cluster awareness`. See the terminology here: https://github.com/yugabyte/jdbc-yugabytedb